bin/GoogleAdsCleanupServices.php

```diff
@@ -61,7 +61,7 @@ class GoogleAdsCleanupServices {
 	 * @param Event|null  $event Composer event.
 	 * @param string|null $path Path of the Ads library.
 	 */
-	public function __construct( Event $event = null, string $path = null ) {
+	public function __construct( ?Event $event = null, ?string $path = null ) {
```

Comment:
I see we added ? over several areas in the codebase. However, I see some of them not being changed.

For instance: https://github.com/woocommerce/google-listings-and-ads/blob/develop/src/API/Google/MerchantReport.php#L67

Reply:
I'm hoping I didn't miss any as I mostly focused on getting the unit tests to pass. So if I did miss something that wasn't unit tested then please do let me know.

The line you linked to doesn't need any change because the function is defined as:
```
public function get_product_view_report( $next_page_token = null ): array {
```

While the parameter is set to null, it doesn't have any type defined, so it doesn't matter if it's a string or null and even PHP 8.4 won't complain about it. If we would write that function again from scratch we should probably define it as:
```
public function get_product_view_report( ?string $next_page_token = null ): array {
```

But I don't think we want to go back through all our code and require types on every parameter.

For the places where we did fix it, we specifically used a `?` as that was introduced in PHP 7.1, we could have also gone for `string|null` but that's only supported in PHP 8.0, [see here](https://php.watch/versions/8.4/implicitly-marking-parameter-type-nullable-deprecated).